linc_convert/utils/spool.py

```diff
@@ -0,0 +1,198 @@
+# stdlib
+import configparser
+import io
+import os
+import warnings
+from glob import glob
+
+import numpy as np
+from scipy.io import loadmat
+
+from linc_convert.utils.math import ceildiv
+
+
+class SpoolSetInterpreter:
+
+    def __init__(self, spool_set_path, info_file=None):
+        if os.path.isfile(spool_set_path) and \
+                os.path.splitext(spool_set_path)[-1] == '.zip':
+            raise NotImplementedError("Zip storage is not supported.")
+            pass
+        elif os.path.isdir(spool_set_path):
+            self.type = 'dir'
+            self.parent = spool_set_path
+            self.file_list = glob(os.path.join(spool_set_path, '*'))
+            self.spool_set = tuple([os.path.split(x)[-1] for x in self.file_list])
+        else:
+            assert False, 'The input data structure is not a ZIP file or Directory'
+
+        self._what_spool_format()
+        self.spool_files = tuple(self._get_spool_names_in_order())  # In order
+        self._get_acquisitionparameters_str()
+        self._get_config()
+        self._extract_config_values()
+
+        if info_file is not None:
+            if os.path.isfile(info_file):
+                self._load_info_file(info_file)
+            else:
+                warnings.warn('ERROR: Info file not found.')
+        self.assembled_spool_shape = (self.numDepths,
+                                      self.spool_shape[2],
+                                      self.spool_shape[0]*len(self.spool_files),
+                                      )
+    def _load_info_file(self, info_file):
+        loaded_info = loadmat(info_file)
+        info = loaded_info.get("info", None)
+        if info is None:
+            warnings.warn("ERROR: 'info' structure not found in the MAT file.")
+            return None
+        try:
+            num_total_frames = int(info['camera'][0][0]['kineticSeriesLength'])
+            num_bg_frames = int(info['camera'][0][0]['backgroundFramesNum'])
+
+        except (KeyError, IndexError, ValueError) as e:
+            warnings.warn(
+                f"ERROR: Unable to extract frame information from info file. {e}")
+            return None
+        num_frames_per_spool = int(self.config['multiimage']['ImagesPerFile'])
+        num_frames_to_load = num_total_frames - num_bg_frames
+        num_frames_to_load = ceildiv(num_frames_to_load, num_frames_per_spool)
+        self.spool_files = self.spool_files[:num_frames_to_load]
+
+    def _make_filename_from_spool_set(self, spool_entry):
+        return os.path.join(self.parent, spool_entry)
+
+    @property
+    def entries(self):
+        if self.type == '.zip':
+            return self.spool_set.entries
+        elif self.type == 'dir':
+            return self.spool_set
```

Comment:
Since we are only supporting `dir`.

```suggestion
        return self.spool_set
```